feat #6 - 맛집 목록, 맛집 상세조회 API 구현

nswon opened the pull request:
### 변경사항
**AS-IS**
- 맛집 목록 조회
- 맛집 상세 조회

**TO-BE**
테스트코드

### 테스트
- [ ] 테스트 코드
- [ ] API 테스트

nswon committed:
feat #6 - 맛집 목록, 맛집 상세조회 API 구현

jjungyeun commented on src/main/java/com/wanted/teamV/repository/QuerydslRestaurantRepository.java:
```diff
@@ -0,0 +1,38 @@
+                .fetch();
+    }
+
+    private BooleanExpression filterRestaurantType(String name) {
```
String 그대로 코드에 넣기보다는 Enum 클래스 정의해서 쓰면 좋을 것 같아요!

nswon replied:
넵 수정하겠습니다

jjungyeun commented on src/main/java/com/wanted/teamV/service/impl/RestaurantServiceImpl.java:
```diff
@@ -1,21 +1,77 @@
+        Restaurant restaurant = restaurantRepository.getById(restaurantId);
+
+        return restaurant.getRatingList().stream()
+                .sorted(Comparator.comparing(Rating::getCreatedAt).reversed())
```
Restaurant 엔티티 조회 후에 연결된 엔티티의 필드를 사용하면, 사용할 때 쿼리가 따로 나가서 아마 쿼리 개수가 많아지는걸로 알고 있어요..! restaurant 조회 시에 fetch join으로 Rating 정보도 같이 조회해오는 것도 좋을 것 같아요

nswon replied:
오 좋은 방법인 것 같아요! 수정하겠습니다

nswon committed:
pr리뷰대로 수정
- 맛집종류 필터링 로직에서 문자열을 enum으로 변경
- 맛집 상세조회에서 그냥 참조된 평가정보를 들고옴 (쿼리 한번 더 나감) -> 맛집정보를 가져올때 fetch join으로 평가정보도 함께 가져옴

nswon committed:
사소한거 수정

nswon committed:
restaurant repository 이름 변경
